lib/rules/color-hex-length/index.cjs

```diff
@@ -65,12 +65,12 @@ const rule = (primary) => {
 					setDeclarationValue(decl, parsedValue.toString());
 				};
 
-				const index = declarationValueIndex(decl) + node.sourceIndex;
-				const endIndex = index + node.value.length;
+				const index = declarationValueIndex(decl) + 1 + node.sourceIndex;
+				const endIndex = index + node.value.length - 1;
 
 				report({
 					message: messages.expected,
-					messageArgs: [hexValue, expectedHex],
+					messageArgs: [hexValue.substring(1), expectedHex.substring(1)],
```

Comment:
I don't fully understand why this is being changed.
Is this because `#` is present before and after the fix?

I think it is fine if we include `#` in both the reported range and the message. Only having the hex values, without the leading `#` is less clear.

I also think that this is less correct on a technical level.
`#fff` is a single token and a single component value in CSS.

Reply:
see https://github.com/stylelint/stylelint/pull/8008#issuecomment-2374012907